File: python/2007/ps-i-love-you-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add P.S. I Love You to the database
movies.insert(
    title="P.S. I Love You", 
    year=2007, 
    plot="A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.", 
    rating=7.1
)

# Confirm that the movie was added
movie = movies.select(
    title="P.S. I Love You", 
    year=2007
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2007/ps-i-love-you-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "P.S. I Love You" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "P.S. I Love You",
                year: 2007,
                plot: "A young widow discovers that her late husband has left her 10 messages intended to help ease her pain and start a new life.",
                rating: 7.1
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "P.S. I Love You",
                year: 2007
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
